gossip/src/crds.rs

```diff
@@ -56,6 +57,22 @@ use {
 const CRDS_SHARDS_BITS: u32 = 12;
 // Number of vote slots to track in an lru-cache for metrics.
 const VOTE_SLOTS_METRICS_CAP: usize = 100;
+// Required number of trailing zero bits for crds signature to get reported to influx
+// mean new push messages received per minute per node
+//      testnet: ~500k,
+//      mainnet: ~280k
+// target: 1-2 signatures reported per minute
+// log2(250k) = ~17.9.
+const SIGNATURE_SAMPLE_TRAILING_ZEROS: u32 = 18;
+
+/// check if last SIGNATURE_SAMPLE_TRAILING_ZEROS bits of signature are 0
+fn should_report_message_signature(signature_bytes: Signature) -> bool {
+    let (trailing_signature_bytes, _) = signature_bytes
+        .as_ref()
+        .split_at(std::mem::size_of::<u64>());
+    u64::from_le_bytes(trailing_signature_bytes.try_into().unwrap()).trailing_zeros()
+        >= SIGNATURE_SAMPLE_TRAILING_ZEROS
```

Comment:
`split_at` might panic; better to avoid the other `unwrap` here as well.
Something like:

```rust
#[inline]
fn should_report_message_signature(signature: &Signature) -> bool {
    let Some(Ok(bytes)) = signature.as_ref().get(..8).map(<[u8; 8]>::try_from) else {
        return false;
    };
    u64::from_le_bytes(bytes).trailing_zeros() >= SIGNATURE_SAMPLE_TRAILING_ZEROS
}
```

or 

```rust
#[inline]
fn should_report_message_signature(signature: &Signature) -> bool {
    Some(SIGNATURE_SAMPLE_TRAILING_ZEROS)
        <= signature
            .as_ref()
            .get(..8)
            .map(<[u8; 8]>::try_from)
            .and_then(Result::ok)
            .map(u64::from_le_bytes)
            .map(u64::trailing_zeros)
}
```

Reply:
do we avoid `unwrap` for same reason as `split_at` aka it can panic?

Reply:
yes